elasticsearch/file/src/main/java/nl/knaw/huc/service/HeaderLinkUtil.java

```diff
@@ -0,0 +1,25 @@
+package nl.knaw.huc.service;
+
+import javax.ws.rs.BadRequestException;
+import javax.ws.rs.core.Link;
+import java.util.UUID;
+import java.util.regex.Pattern;
+
+public class HeaderLinkUtil {
+
+  // Rough approximation of UUID:
+  private static final Pattern UUID_PATTERN = Pattern.compile(".*([0-9a-f-]{36}).*");
+
+  /**
+   * Extract file UUID from header link
+   */
+  public static UUID extractUuid(String headerLinkValue) {
+    var url = Link.valueOf(headerLinkValue).getUri().toString();
+    var matcher = UUID_PATTERN.matcher(url);
+    if (!matcher.find()) {
+      throw new BadRequestException("Header Link does not contain file UUID: " + headerLinkValue);
+    }
+    return UUID.fromString(matcher.group(1));
```

Comment:
als het UUID.fromString() niet lukt, krijg je hier een IllegalArgumentException -> TextRepoRequestException -> 503 treintje

Reply:
Waarom resulteert een BadRequestException in een TextRepoRequestException?

Reply:
Ik dacht dat de TextRepoRequestException (en -Mapper) ervoor zorgen dat een IllegalArgumentException een TextRepoRequestException wordt. Daarom kun je dan handig in een Resource een IAE gooien en wordt het een TRRE.
In dit geval gooit UUID.fromString in zijn implementatie een IAE als het hem niet lukt het UUID te parseren.